### PR TITLE
Quickcheck

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Added automated tests using quickcheck. Found a bug.

## Changes
- `pub fn new` now takes a page size
- `pub fn default` was added, allocates 1mb pages by default
- added quickcheck tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,11 +3,9 @@ rust:
   - nightly
 
 before_script: |
-  rustup component add rustfmt-preview &&
-  cargo install clippy -f
+  rustup component add rustfmt-preview
 script: |
   cargo fmt -- --write-mode=diff &&
-  cargo clippy -- -D clippy &&
   cargo build --verbose &&
```
Comment (ralphtheninja) [inline .travis.yml]: I was actually wondering about these lines :wink:

File: src/lib.rs [hunk 2]
```diff
@@ -1,21 +1,9 @@
...
-//! let mut file = ram::Sync::new();
-//! file.write(0, b"hello")?;
-//! file.write(5, b" world")?;
-//! let text = file.read(0, 11)?;
-//! assert_eq!(text, b"hello world");
-//! ```
+#![deny(missing_docs)]
+#![cfg_attr(test, deny(warnings))]
+#![feature(external_doc)]
+#![doc(include = "../README.md")]
```
Comment (ralphtheninja) [inline src/lib.rs]: :+1:

File: tests/sync_regression.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+extern crate random_access_memory as ram;
```
Comment (ralphtheninja) [inline tests/sync_regression.rs]: What's the idea of this file? Just a separation of tests into api tests and bug tests?

Comment (yoshuawuyts) [reply tests/sync_regression.rs]: yeah, pretty much!

When a test fails you can re-run it with `cargo test --test <filename>`, which also made testing for these specific regressions locally a lot faster. Also really like writing a postmortem, so it becomes clear which mistakes we've made in the past - as some of these random inputs can get kinda weird over time.

Eventually we should introduce stuff like failpoints into this https://github.com/pingcap/fail-rs, so we can check how writes persist between shutdowns / random crashes. Things might get super weird at that point, so it'll be good to check!

**Pull request merged**